### PR TITLE
Merge feature/msbuild to dev

### PR DESCRIPTION
**Pull request by natemcmaster**
We only have two repos left on 'dev' EF and MVC. I'll pin their build.ps1 before we merge this.

This deprecates support for building project.json with KoreBuild.

### CODE REVIEW

File: build/KoreBuild.ps1 [hunk 1]
```diff
@@ -42,8 +54,9 @@ if ($env:KOREBUILD_SKIP_RUNTIME_INSTALL -eq "1")
 }
 else
 {
+    # Install the version of dotnet-cli used to compile
     & "$koreBuildFolder\dotnet\dotnet-install.ps1" -Channel $dotnetChannel -Version $dotnetVersion -Architecture x64
-    InstallSharedRuntime '1.0.0' 'preview'
```
Comment (pranavkm) [inline build/KoreBuild.ps1:46]: Do none of our tools need 1.0.0 anymore?

File: build/KoreBuild.ps1 [hunk 2]
```diff
@@ -42,26 +63,26 @@ if ($env:KOREBUILD_SKIP_RUNTIME_INSTALL -eq "1")
...
-    if ($env:KOREBUILD_DOTNET_SHARED_RUNTIME_VERSION)
-    {
-        $channel = 'master'
-        if ($env:KOREBUILD_DOTNET_SHARED_RUNTIME_CHANNEL)
-        {
-            $channel = $env:KOREBUILD_DOTNET_SHARED_RUNTIME_CHANNEL
-        }
-        InstallSharedRuntime $env:KOREBUILD_DOTNET_SHARED_RUNTIME_VERSION $channel
-    }
+    # TODO restore installing pre-release versions of sharedfx when we upgrade to .NET Core 2.0
```
Comment (pranavkm) [inline build/KoreBuild.ps1]: No TODOs or commented out code.

File: build/KoreBuild.ps1 [hunk 3]
```diff
@@ -73,41 +94,37 @@ if (!($env:Path.Split(';') -icontains $dotnetLocalInstallFolder))
 $sharedPath = (Join-Path (Split-Path ((get-command dotnet.exe).Path) -Parent) "shared");
 (Get-ChildItem $sharedPath -Recurse *dotnet.exe) | %{ $_.FullName } | Remove-Item;
 
-if (!(Test-Path "$koreBuildFolder\Sake"))
+# We still nuget because dotnet doesn't have support for pushing packages
+# TODO remove. dotnet nuget push now exists
```
Comment (pranavkm) [inline build/KoreBuild.ps1]: Let's file a bug and remove the todo.

File: build/KoreBuild.sh [hunk 4]
```diff
@@ -61,14 +91,15 @@ else
...
-    $koreBuildFolder/dotnet/dotnet-install.sh --channel preview --version $version
-    install_shared_runtime '1.0.0' 'preview'
-    install_shared_runtime $sharedRuntimeVersion 'master'
-    if [ ! -z "$KOREBUILD_DOTNET_SHARED_RUNTIME_VERSION" ]; then
-        channel="$KOREBUILD_DOTNET_SHARED_RUNTIME_CHANNEL"
-        [ -z "$channel" ] && channel="master"
-        install_shared_runtime $KOREBUILD_DOTNET_SHARED_RUNTIME_VERSION $channel
-    fi
+    $koreBuildFolder/dotnet/dotnet-install.sh --channel $KOREBUILD_DOTNET_CHANNEL --version $KOREBUILD_DOTNET_VERSION
+    # TODO restore installing pre-release versions of sharedfx when we upgrade to .NET Core 2.0
```
Comment (pranavkm) [inline build/KoreBuild.sh]: ditto

File: build/NuGet.config [hunk 5]
```diff
@@ -3,4 +3,4 @@
   <packageSources>
     <add key="aspnetcore-tools" value="https://dotnet.myget.org/F/aspnetcore-tools/api/v3/index.json" />
   </packageSources>
-</configuration>
+</configuration>
```
Comment (pranavkm) [inline build/NuGet.config]: Undo

File: build/dotnet/dotnet-install.ps1 [hunk 6]
```diff
@@ -45,20 +45,24 @@
 .PARAMETER Verbose
```
Comment (pranavkm) [inline build/dotnet/dotnet-install.ps1:45]: Seems unrelated to the msbuild conversion. Make this a separate commit

File: build/targets/standard-lifecycle.targets [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+Items: ProjectsToTest. Defaults to test/*/*.csproj
+Properties: ExcludeFromTest. ItemSpec for projects to avoid testing.
+
+Runs the VSTest on all projects in the ProjectsToTest itemgroup.
+###################################################################
+-->
+
+  <PropertyGroup>
+    <!-- TODO if VS Test doesn't make minimal output the default log setting, we can set that here. cref https://github.com/Microsoft/vstest/issues/301 -->
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: This is fixed. Remove

File: build/targets/standard-lifecycle.targets [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+<!--
+###################################################################
+Target: PackageProjects
+
+Executes /t:Pack on all projects matching src/*/*.csproj.
+###################################################################
+-->
+  <Target Name="PackageProjects">
+    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution.
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: Tracking issue + Remove todo

File: build/targets/standard-lifecycle.targets [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+###################################################################
+Convenience targets.
+
+These are not part of the standard lifecycle but are useful
+###################################################################
+-->
+
+  <Target Name="Rebuild" DependsOnTargets="RebuildSolutions" />
+
+  <!-- TODO implement. See https://github.com/aspnet/BuildTools/issues/162 -->
```
Comment (pranavkm) [inline build/targets/standard-lifecycle.targets]: Remove the target since it's not doing anything

File: build/KoreBuild.ps1 [hunk 10]
```diff
@@ -73,41 +94,37 @@ if (!($env:Path.Split(';') -icontains $dotnetLocalInstallFolder))
...
-    $makeFilePath = "$koreBuildFolder\shade\makefile.shade"
+    mkdir $msbuildArtifactsDir | Out-Null
 }
+$msBuildArguments | Out-File -Encoding ASCII -FilePath $msBuildResponseFile
 
-Write-Host "Using makefile: $makeFilePath"
-
-$env:KOREBUILD_FOLDER=$koreBuildFolder
-&"$koreBuildFolder\Sake\0.2.2\tools\Sake.exe" -I $koreBuildFolder\shade -f $makeFilePath @args
+exec dotnet build `@"$msBuildResponseFile"
```
Comment (pranavkm) [inline build/KoreBuild.ps1]: Add a trailing new line to these files. There's a bunch missing these

**Review by pranavkm — CHANGES_REQUESTED** [review]

**Comment by natemcmaster** [comment]
Will do this later. Too many people still using 'dev' branch.

**Comment by natemcmaster** [comment]
MVC merged sooner than expected. Now all Universe repos are on feature/msbuild. Let's move this to dev.

File: build/targets/standard-lifecycle.targets (hunk 8 above)
Comment (natemcmaster) [reply build/targets/standard-lifecycle.targets]: https://github.com/aspnet/KoreBuild/issues/184

File: build/KoreBuild.ps1 (hunk 1 above)
Comment (natemcmaster) [reply build/KoreBuild.ps1:46]: A few of our tools do, but the 1.0.0-rc4 SDK installs both the 1.0.4 and 1.1.1 runtimes automatically

File: build/dotnet/dotnet-install.ps1 (hunk 6 above)
Comment (natemcmaster) [reply build/dotnet/dotnet-install.ps1:45]: These changes reflect updates and bug fixes to the dotnet-install.ps1 script in https://github.com/dotnet/cli. The layout of assets on dotnetcli.azureedge.net is slightly different for the MSBuild SDK than it was for project.json SDK.

**Comment by natemcmaster** [comment]
🆙 :date: added changes per PR feedback

**Comment by natemcmaster** [comment]
Before we merge this, I've tagged the current version of 'dev' as "legacy-build". I'm updating any other aspnet repos to use this so that merging this MSBuild work shouldn't break them.

This intentionally means that anyone using "legacy-build" won't receive updates as we're no longer updating our project.json scripts.

**Review by analogrelay — APPROVED** [review]
Looks good to me

**Pull request merged**